espnet2/bin/spk_embed_extract.py

```diff
@@ -0,0 +1,693 @@
+        )
+    else:
+        # Suppress logging if RANK != 0
+        logging.basicConfig(
+            level="ERROR",
+            format=f"[{os.uname()[1].split('.')[0]}"
+            f":{distributed_option.dist_rank}/{distributed_option.dist_world_size}]"
+            f" %(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s",
+        )
+    # Invoking torch.distributed.init_process_group
+    distributed_option.init_torch_distributed()
+
+    logging.basicConfig(
+        level=args.log_level,
+        format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s",
+    )
+
+    if args.ngpu >= 1:
+        device = "cuda"
+    else:
+        device = "cpu"
+
+    # 1. Set random-seed
+    set_all_random_seed(args.seed)
+
+    # 2. define train args
+    spk_model, spk_train_args = SpeakerTask.build_model_from_file(
+        args.spk_train_config, args.spk_model_file, device
+    )
+
+    # 3. Overwrite args with inference args
+    args = vars(args)
+    args["valid_data_path_and_name_and_type"] = args["data_path_and_name_and_type"]
+    args["valid_shape_file"] = args["shape_file"]
+    args["preprocessor_conf"] = {
+        "target_duration": args["target_duration"],
+        "num_eval": args["num_eval"],
+        "noise_apply_prob": 0.0,
+        "rir_apply_prob": 0.0,
+    }
+
+    merged_args = vars(spk_train_args)
+    merged_args.update(args)
+    args = argparse.Namespace(**merged_args)
+
+    # 4. Build data-iterator
+    distributed_option.distributed = False
+    iterator = SpeakerTask.build_iter_factory(
+        args=args,
+        distributed_option=distributed_option,
+        mode="valid",
+    )
+    distributed_option.distributed = True
+    loader = iterator.build_iter(0)
+
+    trainer_options = SpeakerTask.trainer.build_options(args)
+    reporter = Reporter()
+
+    # 5. Run inference for EER and minDCF calculation
+    with reporter.observe("valid") as sub_reporter:
+        SpeakerTask.trainer.extract_embed(
+            model=spk_model,
+            iterator=loader,
+            reporter=sub_reporter,
+            options=trainer_options,
+            distributed_option=distributed_option,
+            output_dir=args.output_dir,
+            average=args.average_embd,
+        )
+
+    if distributed_option.distributed:
+        torch.distributed.barrier()
+    if not distributed_option.distributed or distributed_option.dist_rank == 0:
+        # logging.info(reporter.log_message())
+
+        # Combine dictionaries into one
+        npzs = glob(args.output_dir + "/embeddings*.npz")
+        logging.info(f"{npzs}")
+        embd_dic = {}
+        for npz in npzs:
+            tmp_dic = dict(np.load(npz))
+            embd_dic.update(tmp_dic)
+        set_name = args.data_path_and_name_and_type[0][0].split("/")[-2]
+        np.savez(args.output_dir + f"/{set_name}_embeddings", **embd_dic)
+        for npz in npzs:
+            os.remove(npz)
+
+
+def get_parser():
```

Comment:
Some arguments are not necessary here, and we can just remove them.

Reply:
actually these are added to make conf/decode.yaml to be useable